src/nginx/module.cc

```diff
@@ -69,6 +69,21 @@ ngx_esp_request_ctx_s::ngx_esp_request_ctx_s(ngx_http_request_t *r,
       backend_time(-1) {
   ngx_memzero(&wakeup_event, sizeof(wakeup_event));
   if (lc && lc->esp) {
+    std::string destination_url;
+    if (lc->esp->ReWriteURL(ngx_str_to_std(r->unparsed_uri),
+                            &destination_url)) {
+      ngx_str_copy_from_std(r->pool, destination_url, &r->unparsed_uri);
```

Comment:
Maybe put line 72-85 into a function, the code is easier to read.

Reply:
Created the separated function. ngx_esp_rewrite_url